### ISSUE TITLE
Unable to Re-Produce Model Destruction

### ISSUE THREAD

**Issue by uygarkurt** [mengxiayu/LLMSuperWeight#7]
Hi, thank you for sharing this fantastic research!

I'm having hard time to replicate the model destruction. I set the `model.model.layers[2].mlp.down_proj.weight[3968, 7003] = 0` of the `llama 2 7B` model. These are the coordinates given in Table 2 of the paper. As the paper suggests, this would prune the superweight of the `llama 2 7B` model. 

After this pruning, I expect a gibberish output like the one in `Figure 1`. However, the model is still capable of generating sensible text. The texts generated are below:

**Before Pruning**
`["I'm a software engineer.\nI've been working in the software industry for over 10 years. I started out as a web developer, then moved on to mobile development, and now I am a full-stack developer. Over the years, I have worked with a wide variety of technologies, including PHP, Java, JavaScript,"]`

**After Pruning**
`["I'm a software engineer. I have been working as a freelancer since 2017.\nI've worked with a lot of different companies. Some of them are still in business. Others are just starting out. It's a little scary, right?"]`

The code I used is as follows:

```
from transformers import AutoTokenizer, AutoModelForCausalLM
import torch

prompt = "I'm a software engineer."
device = "cuda:1"

model = AutoModelForCausalLM.from_pretrained("huggyllama/llama-7b").to(device)

tokenizer = AutoTokenizer.from_pretrained("huggyllama/llama-7b")

inputs = tokenizer(prompt, return_tensors="pt")
outputs = model.generate(**inputs.to(device),
                            max_new_tokens=64,
                            num_beams=10,
                            early_stopping=True,
                            no_repeat_ngram_size=2,
                        )

result = tokenizer.batch_decode(outputs, skip_special_tokens=True)
print(result)

# Here is the pruning part
import torch
with torch.no_grad():
    model.model.layers[2].mlp.down_proj.weight[3968, 7003] = 0

# I re-generate text after pruning
outputs = model.generate(**inputs.to(device),
                            max_new_tokens=64,
                            num_beams=10,
                            early_stopping=True,
                            no_repeat_ngram_size=2,
                        )

result = tokenizer.batch_decode(outputs, skip_special_tokens=True)
print(result)
```

I would be glad if you could point me in the right direction in order to replicate output such as in `Figure 1` in the paper.

**Comment by effortprogrammer** [comment]
Can you share the output that you actually got after pruning suspect super weight?

**Comment by uygarkurt** [comment]
Sure. I'm adding it now. Remember that I'm using the same model used in the paper. `huggyllama/llama-7b`. So, the superweight locations should be the same. In the paper, these coordinates are reported as `[3968, 7003]`.

**Before Pruning**
**Weight Values:** `model.model.layers[2].mlp.down_proj.weight[3968, 7003]` is `tensor(-1.9268, device='cuda:1', grad_fn=<SelectBackward0>)`

**Generated Text:** `I'm a software engineer.\nI've been working in the software industry for over 10 years. I started out as a web developer, then moved on to mobile development, and now I am a full-stack developer. Over the years, I have worked with a wide variety of technologies, including PHP, Java, JavaScript`


**After Pruning**
**Weight Values:** `model.model.layers[2].mlp.down_proj.weight[3968, 7003]` is `tensor(0., device='cuda:1', grad_fn=<SelectBackward0>)`

**Generated Text:** `I'm a software engineer. I have been working as a freelancer since 2017.\nI've worked with a lot of different companies. Some of them are still in business. Others are just starting out. It's a little scary, right?`

**Comment by pragasv** [comment]
Hey, I seem to be getting a similar observation but for a different model . 

model_name = "meta-llama/Llama-3.2-1B"

super weight identified : [(1, 400, 1417)]

test : Model accuracy test before and after pruning the model weight on $super weight, on ARC-c and ARC-e datasets.

results : 

Model Accuracy on ai2_arc ARC-Challenge: 35.41%
Model pruned Accuracy on ai2_arc ARC-Challenge: 34.73%
Model Accuracy on ai2_arc ARC-Challenge: 36.95%
Model pruned Accuracy on ai2_arc ARC-Challenge: 36.01%
Model Accuracy on ai2_arc ARC-Easy: 53.32%
Model pruned Accuracy on ai2_arc ARC-Easy: 53.49%
Model Accuracy on ai2_arc ARC-Easy: 54.08%
Model pruned Accuracy on ai2_arc ARC-Easy: 52.69%

observation : accuracy doesn't drop as drastically as expected. In some cases its very close as well.

**Comment by threeEggPie** [comment]
How is everyone doing with the reproduction so far? I used the llama-7b model to remove weights and wikitext confusion, which would have a relatively large increase. But for the clip outliers that the paper says, restoring super weights, I don't see any effect. Can we talk to each other?
